src/main/java/io/reactivesocket/cli/Main.java

```diff
@@ -135,6 +173,43 @@ public void run() throws IOException, URISyntaxException, InterruptedException {
         }
     }
 
+    public Payload parseSetupPayload() {
+        return new Payload() {
+            @Override
+            public ByteBuffer getMetadata() {
+                return Frame.NULL_BYTEBUFFER;
+            }
+
+            @Override
+            public ByteBuffer getData() {
+                String source = null;
+
+                if (setup.startsWith("@")) {
+                    try {
+                        source = Files.asCharSource(setupFile(), Charsets.UTF_8).read();
```

Comment:
This can be done before creating the `Payload` instance, which makes the validation eager too.